app/controllers/publish/courses/age_range_controller.rb

```diff
@@ -87,12 +85,24 @@ def current_step
         :age_range
       end
 
+      def build_provider
+        return if params[:provider_code].blank?
+
+        @provider = @recruitment_cycle.providers.find_by!(
+          provider_code: params[:provider_code].upcase,
+        )
+      end
+
+      def build_recruitment_cycle
+        @recruitment_cycle = RecruitmentCycle.find_by(
+          year: params[:recruitment_cycle_year],
+        ) || RecruitmentCycle.current_recruitment_cycle
+      end
+
       def build_course
```

Comment:
Can be dropped and as it should be using

https://github.com/DFE-Digital/teacher-training-api/blob/db1ca9b0b3a8c8d999e544cf1028766f1814b47b/app/models/concerns/publish/course_basic_detail_concern.rb#L60-L62

Reply:
Have to call pundit though? 
```
 Pundit::AuthorizationNotPerformedError:
       Publish::Courses::AgeRangeController
```
Something like this?
```
def build_course
      #  @course = @provider.courses.find_by!(course_code: params[:code].upcase)
        super
        authorize @course
      end
```

Reply:
I have added the permissions in concern directly. eg https://github.com/DFE-Digital/teacher-training-api/pull/2566/files#diff-b6451eb1b3fe8ddfef71d842610dc538ff9a6e6fdb6cbfef6eda1fde91d6b02dR21. Maybe you could do the same and whoevers goes in will take precedence. Think you need to authorise the provider not the course?

Reply:
I think the `before_action` is not needed as it called via `super`